### ISSUE TITLE
Rule Change: `no-else-return` should catch `break`/`continue` statements

### ISSUE THREAD

**Issue by B2o5T** [eslint/eslint#15496]
### What rule do you want to change?

no-else-return

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
for (const item of list) {
  if (foo) {
    if (a) {
      break
    } else {
      return
    }
  }

  if (bar) {
    if (a) {
      continue
    } else {
      return
    }
  }
}
```

### What does the rule currently do for this code?

no warnings

### What will the rule do after it's changed?

should propose to remove unnecessary `else` because:
- `break` statement terminates the current loop
- `continue` statement terminates the execution of the statements in the current iteration

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

Expected output
```js
for (const item of list) {
  if (foo) {
    if (a) {
      break
    }
    return
  }

  if (bar) {
    if (a) {
      continue
    }
    return
  }
}
```

**Comment by nzakas** [comment]
Thanks for the suggestion. I verified that we currently don’t catch this case, though I’m not sure if that’s a bug or not (it seems likely that it’s a bug).

@eslint/eslint-tsc thoughts on if we should treat this like a bug instead of an enhancement?

**Comment by mdjermanovic** [comment]
I think this isn't a bug because this rule disallows `else` after _`return`_ (hence the name `no-else-return`, although the word order is confusing).

There was an accepted but never implemented proposal for a new rule `no-useless-else` that would include `break`, `continue`, and `throw` as well.

Given our current policies about stylistic rules, I think the best course of action would be to implement `no-useless-else` in a plugin.

**Comment by mdjermanovic** [comment]
> There was an accepted but never implemented proposal for a new rule `no-useless-else`

I forgot to include the link: https://github.com/eslint/eslint/issues/8028

**Comment by nzakas** [comment]
Good call. As we already decided this behavior wasn't a bug, I'm closing this issue.
